app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,22 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
```

Comment:
` ||= ` giải thích code này giúp em

Reply:
toán tử này này gần như tương đương define? Toán tử này hiểu là kiểm trả giá trị của biến có đúng hay ko .Nếu tìm dc 1 ông user với id lấy từ session thì sẽ gắn cho biến trường current_user. còn không nó sẽ gắn là nil.

Reply:
tào lao, anh tìm hiểu lại cú pháp này
nó là viết ngắn gọn của if-else đấy

Reply:
thì ý anh là vậy nếu biểu thức vế 2 trả về true thì gắn current_user bằng ông user với id đó. còn không là else nó gắn nil mà. anh hiểu thế không biết đúng không